src/sagemaker_sklearn_extension/feature_extraction/sequences.py

```diff
@@ -0,0 +1,193 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#      http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_array, check_is_fitted
+from tsfresh import extract_features
+from tsfresh.feature_extraction import ComprehensiveFCParameters
+from tsfresh.utilities.dataframe_functions import impute
+
+from sagemaker_sklearn_extension.preprocessing.data import RobustStandardScaler
+
+
+class TSFlattener(BaseEstimator, TransformerMixin):
+    """Convert lists of strings of varying length into an np.array.
+
+    The input is a collection of lists of strings, with each string containing a sequence of comma-separate numbers.
+    TSFlattener extracts the numerical values contained in these strings and pads shorter sequences (if any) with NaNs,
+    so that all sequences match the length of the longest sequence. TSFlattener returns a single np.array as output.
+
+    Any missing values (represented either by a NaN, inf or empty string) are converted to np.nans.
```

Comment:
what happens to entire entries that cannot be parsed?
`Input = [["1,, 3, 44"],      None,       ["11, 111"],             ["NaN, , 1, NaN"]]`
`output=?`

Can you add a comment about that?

Reply:
These are now also filled with NaNs to match the length of the longest sequence. Added a comment.